service/src/test/org/apache/hive/service/auth/ldap/TestLdapSearch.java

```diff
@@ -290,4 +290,50 @@ public void testIsUserMemberOfGroupWhenUserDn() throws NamingException {
     assertTrue(search.isUserMemberOfGroup("CN=User1,OU=org1,DC=foo,DC=bar", "grp1"));
     assertFalse(search.isUserMemberOfGroup("CN=User2,OU=org1,DC=foo,DC=bar", "grp2"));
   }
+  @Test
+  public void testExecuteUserAndGroupFilterQueryWithSpecialCharacter() throws NamingException {
+    final String groupSearchFilter = "member=CN={0},OU=org1,DC=foo,DC=bar";
+    final String groupBaseDn = "dc=example,dc=com";
+    NamingEnumeration<SearchResult> validResult1 = LdapTestUtils.mockNamingEnumeration("Test User 1");
+    NamingEnumeration<SearchResult> validResult2 = LdapTestUtils.mockNamingEnumeration("Test User 2");
+    NamingEnumeration<SearchResult> validResult3 = LdapTestUtils.mockNamingEnumeration("Test User 3");
+    NamingEnumeration<SearchResult> validResult4 = LdapTestUtils.mockNamingEnumeration("Test User 4");
+    NamingEnumeration<SearchResult> validResult5 = LdapTestUtils.mockNamingEnumeration("Test User 5");
+
+    when(ctx.search(anyString(), contains("member=CN=Test \\5C User,OU=org1,DC=foo,DC=bar,OU=org1,DC=foo,DC=bar"), any(SearchControls.class))).thenReturn(validResult1);
+    when(ctx.search(anyString(), contains("member=CN=Test \\2A User,OU=org1,DC=foo,DC=bar,OU=org1,DC=foo,DC=bar"), any(SearchControls.class))).thenReturn(validResult2);
+    when(ctx.search(anyString(), contains("member=CN=Test \\28 User,OU=org1,DC=foo,DC=bar,OU=org1,DC=foo,DC=bar"), any(SearchControls.class))).thenReturn(validResult3);
+    when(ctx.search(anyString(), contains("member=CN=Test \\29 User,OU=org1,DC=foo,DC=bar,OU=org1,DC=foo,DC=bar"), any(SearchControls.class))).thenReturn(validResult4);
+    when(ctx.search(anyString(), contains("member=CN=Test \\00 User,OU=org1,DC=foo,DC=bar,OU=org1,DC=foo,DC=bar"), any(SearchControls.class))).thenReturn(validResult5);
+    search = new LdapSearch(conf, ctx);
+
+    // contains \
+    List<String> result = search.executeUserAndGroupFilterQuery("Test , User", "Test \\ User,OU=org1,DC=foo,DC=bar",
+        groupSearchFilter, groupBaseDn);
+    assertEquals(1,result.size());
```

Comment:
nit: according to basic code formatting guidelines, a blank space should appear after commas in argument lists
so:
```
assertEquals(1,result.size());
```
a code reformat in IDE handles this automatically

Reply:
Fixed.